geoh5py/ui_json/parameters.py

```diff
@@ -0,0 +1,491 @@
+        )
+        super().__init__(name, validations=validations, **kwargs)
+
+
+class BoolParameter(FormParameter):
+    base_validations: Validation = {"types": [bool]}
+
+    def __init__(self, name, validations=None, **kwargs):
+        validations = (
+            dict(self.base_validations, **validations)
+            if validations
+            else self.base_validations
+        )
+        super().__init__(name, validations=validations, **kwargs)
+
+
+class IntegerParameter(FormParameter):
+    base_validations: Validation = {"types": [int]}
+    integer_form_validations: dict[str, Validation] = {
+        "min": {"types": [int, type(None)]},
+        "max": {"types": [int, type(None)]},
+    }
+    form_validations: dict[str, Validation] = dict(
+        FormParameter.form_validations, **integer_form_validations
+    )
+    valid_members: list[str] = list(form_validations.keys())
+    identifier_members: list[str] = []
+
+    def __init__(self, name, validations=None, **kwargs):
+        self._min: int | None = None
+        self._max: int | None = None
+        validations = (
+            dict(self.base_validations, **validations)
+            if validations
+            else self.base_validations
+        )
+        super().__init__(name, validations=validations, **kwargs)
+
+
+class FloatParameter(FormParameter):
+    base_validations: Validation = {"types": [float]}
+    float_validations: dict[str, Validation] = {
+        "min": {"types": [float, type(None)]},
+        "max": {"types": [float, type(None)]},
+        "precision": {"types": [int, type(None)]},
+        "line_edit": {"types": [bool, type(None)]},
+    }
+    form_validations: dict[str, Validation] = dict(
+        FormParameter.form_validations, **float_validations
+    )
+    valid_members: list[str] = list(form_validations.keys())
+    identifier_members: list[str] = ["precision", "line_edit"]
+    FormParameter.key_map.update({"lineEdit": "line_edit"})
+
+    def __init__(self, name, validations=None, **kwargs):
+        self._min: float | None = None
+        self._max: float | None = None
+        self._precision: int | None = None
+        self._line_edit: bool | None = None
+        validations = (
+            dict(self.base_validations, **validations)
+            if validations
+            else self.base_validations
+        )
+        super().__init__(name, validations=validations, **kwargs)
+
+
+class ChoiceStringParameter(FormParameter):
+    base_validations: Validation = {"types": [str]}
+    choice_string_validations: dict[str, Validation] = {
+        "choice_list": {"required": True, "types": [list]}
+    }
+    form_validations: dict[str, Validation] = dict(
+        FormParameter.form_validations, **choice_string_validations
+    )
+    valid_members: list[str] = list(form_validations.keys())
+    identifier_members: list[str] = ["choice_list"]
+    FormParameter.key_map.update({"choiceList": "choice_list"})

```

Comment:
Is it not going to alter the base class for all future instances? Maybe we just have one long list as a constant of all the name conversions?

Reply:
it doesn't mess with key map for other instances, but I like the proposed change.  I'll move to the top of the parameters.py file